[Bug] on build with iod-converter

@HamidehGMAO commented on [Thu Sep 02 2021](https://github.com/JCSDA-internal/soca-science/issues/346)

## Current behavior (describe the bug)

I seem to have issue building the bundle with ioda-converter. I get the following error. it builds fine with commenting the ioda-converter build line in the CMakeList.
"
-- Configuring done
CMake Error at /gpfsm/dswdev/jcsda/modules/core/ecbuild/ecmwf-3.6.1/share/ecbuild/cmake/ecbuild_add_test.cmake:438 (add_test):
  Error evaluating generator expression:

    $<TARGET_FILE_DIR:_ioda_python>

  No target "_ioda_python"
Call Stack (most recent call first):
  ioda-converters/test/CMakeLists.txt:560 (ecbuild_add_test)

"



---

@guillaumevernieres commented on [Thu Sep 02 2021](https://github.com/JCSDA-internal/soca-science/issues/346#issuecomment-911639072)

@HamidehGMAO , try that option `-DBUILD_PYTHON_BINDINGS=ON`, if that fails and you don't need the ioda-converters, just comment it out in the cmakelist. 


---

@HamidehGMAO commented on [Thu Sep 02 2021](https://github.com/JCSDA-internal/soca-science/issues/346#issuecomment-911648565)

@guillaumevernieres OK, I will try it. Thanks

---

@HamidehGMAO commented on [Thu Sep 02 2021](https://github.com/JCSDA-internal/soca-science/issues/346#issuecomment-911733445)

Thanks @guillaumevernieres. It worked adding the -DBUILD_PYTHON_BINDINGS=ON option. I'll modify the wiki for SOCA-GEOS

---

@shlyaeva commented on [Thu Sep 02 2021](https://github.com/JCSDA-internal/soca-science/issues/346#issuecomment-911780187)

I've been seeing this issue on mac too. I'm going to reopen and move this issue to ioda-converters to open a discussion on whether we can avoid adding extra options in ecbuild, and still have ioda-converters build.

--- a/CI/buildspec_clang.yml
+++ b/CI/buildspec_clang.yml
@@ -82,7 +82,7 @@ phases:
     on-failure: CONTINUE
     commands:
       - cd /build_container
-      - ecbuild -Wno-dev -DCMAKE_BUILD_TYPE=RelWithDebInfo -DCDASH_OVERRIDE_GIT_BRANCH=$CODEBUILD_GIT_BRANCH -DCTEST_UPDATE_VERSION_ONLY=FALSE -DBUILD_PYTHON_BINDINGS=ON /jcsda/ioda-bundle
+      - ecbuild -Wno-dev -DCMAKE_BUILD_TYPE=RelWithDebInfo -DCDASH_OVERRIDE_GIT_BRANCH=$CODEBUILD_GIT_BRANCH -DCTEST_UPDATE_VERSION_ONLY=FALSE /jcsda/ioda-bundle
       - cd /build_container/iodaconv
       - cp ../DartConfiguration.tcl .
       - sed -i 's/ioda-bundle/ioda-bundle\/iodaconv/' DartConfiguration.tcl

--- a/CI/buildspec_gnu.yml
+++ b/CI/buildspec_gnu.yml
@@ -79,7 +79,7 @@ phases:
         && echo $CC
         && echo $CXX
         && echo $FC
-        && CC=mpicc CXX=mpicxx FC=mpifort ecbuild -Wno-dev -DCMAKE_BUILD_TYPE=RelWithDebInfo -DCDASH_OVERRIDE_GIT_BRANCH=$CODEBUILD_GIT_BRANCH -DCTEST_UPDATE_VERSION_ONLY=FALSE -DENABLE_GPROF=ON -DBUILD_PYTHON_BINDINGS=ON /jcsda/ioda-bundle/"
+        && CC=mpicc CXX=mpicxx FC=mpifort ecbuild -Wno-dev -DCMAKE_BUILD_TYPE=RelWithDebInfo -DCDASH_OVERRIDE_GIT_BRANCH=$CODEBUILD_GIT_BRANCH -DCTEST_UPDATE_VERSION_ONLY=FALSE -DENABLE_GPROF=ON /jcsda/ioda-bundle/"
 
 
       - su - jedi -c "cd /home/jedi/iodaconv

--- a/CI/buildspec_intel.yml
+++ b/CI/buildspec_intel.yml
@@ -88,7 +88,7 @@ phases:
       - cd /home/jedi
       - echo $PYTHONPATH
       - export PYTHONPATH=/usr/local/lib:/usr/local/lib/python3.8/site-packages:$PYTHONPATH
-      - ecbuild -Wno-dev -DCMAKE_BUILD_TYPE=RelWithDebInfo -DCDASH_OVERRIDE_GIT_BRANCH=$CODEBUILD_GIT_BRANCH -DCTEST_UPDATE_VERSION_ONLY=FALSE -DBUILD_PYTHON_BINDINGS=ON /jcsda/ioda-bundle/
+      - ecbuild -Wno-dev -DCMAKE_BUILD_TYPE=RelWithDebInfo -DCDASH_OVERRIDE_GIT_BRANCH=$CODEBUILD_GIT_BRANCH -DCTEST_UPDATE_VERSION_ONLY=FALSE /jcsda/ioda-bundle/
       - cd /home/jedi/iodaconv
       - cp ../DartConfiguration.tcl .
       - sed -i 's/ioda-bundle/ioda-bundle\/iodaconv/' DartConfiguration.tcl

--- a/test/CMakeLists.txt
+++ b/test/CMakeLists.txt
@@ -230,6 +230,7 @@ set(IODA_CONV_COMP_TOL "0.5e-4")
 #        ENVIRONMENT "PYTHONPATH=$<TARGET_FILE_DIR:_ioda_python>/../"
 #        ...
 #        )
+set(IODACONV_PYTHONPATH "$<TARGET_FILE_DIR:_ioda_python>/../:$ENV{PYTHONPATH}")
 
 #===============================================================================
 # Marine converters
@@ -559,7 +560,7 @@ ecbuild_add_test( TARGET  test_${PROJECT_NAME}_viirs_aod
 
 ecbuild_add_test( TARGET  test_${PROJECT_NAME}_tropomi_no2
                   TYPE    SCRIPT
-                  ENVIRONMENT "PYTHONPATH=$<TARGET_FILE_DIR:_ioda_python>/../"
+                  ENVIRONMENT "PYTHONPATH=${IODACONV_PYTHONPATH}"
                   COMMAND ${CMAKE_BINARY_DIR}/bin/iodaconv_comp.sh
                   ARGS    netcdf
                           "${CMAKE_BINARY_DIR}/bin/tropomi_no2_nc2ioda.py

